flows/common/common.go

```diff
@@ -3,12 +3,152 @@ package common
 
 import (
 	"fmt"
+	"sync"
+	"time"
 
 	"github.com/google/gopacket"
+	"github.com/google/gopacket/layers"
+	"github.com/google/gopacket/pcap"
 	"github.com/open-traffic-generator/snappi/gosnappi/otg"
 	"github.com/openconfig/magna/lwotg"
+	"k8s.io/klog"
 )
 
+var (
+	// timeout specifies how long to wait for a PCAP handle.
+	pcapTimeout = 30 * time.Second
+	// packetBytes is the number of bytes to read from an input packet.
+	packetBytes int = 100
+)
+
+// hdrsFunc is a function which specifies which packet headers to expect.
+type hdrsFunc func(*otg.Flow) ([]gopacket.SerializableLayer, error)
+
+// matchFunc is a function which determines if a packet p matchs the headers
+// hdrs.
+type matchFunc func(hdrs []gopacket.SerializableLayer, p gopacket.Packet) bool
+
+// Handler creates a new flow generator function based on the header and match
+// function provided.
+func Handler(fn hdrsFunc, match matchFunc, reporter *Reporter) lwotg.FlowGeneratorFn {
+	return func(flow *otg.Flow, intfs []*lwotg.OTGIntf) (lwotg.TXRXFn, bool, error) {
+		hdrs, err := fn(flow)
+		if err != nil {
+			return nil, false, err
+		}
+
+		fc := NewCounters()
+		fc.Headers = hdrs
+
+		pps, err := Rate(flow, hdrs)
+		if err != nil {
+			return nil, false, fmt.Errorf("cannot calculate rate, %v", err)
+		}
+
+		tx, rx, err := Ports(flow, intfs)
+		if err != nil {
+			return nil, false, fmt.Errorf("cannot determine ports, %v", err)
+		}
+
+		fc.Name = &val{s: flow.Name, ts: flowTimeFn()}
+		klog.Infof("generating flow %s: tx: %s, rx: %s, rate: %d pps", flow.GetName(), tx, rx, pps)
+		reporter.AddFlow(flow.Name, fc)
+
+		genFunc := func(stop chan struct{}) {
+			f := reporter.Flow(flow.Name)
+			klog.Infof("MPLSFlowHandler send function started.")
```

Comment:
Suggest that we replace `MPLSFlowHandler` with the `flow.Name` argument here -- which is probably more useful for logging anyway.

Reply:
oops